### DISCUSSION TITLE
Kratos and Hydra interaction accept oauth2 login request

### DISCUSSION THREAD

**Discussion by MiraBlack** [ory/kratos#3224] [Q&A]
Hello! There was a problem in the process of connecting the hydra along with kratos and integrating them into the existing login page.
I use
- oryd/hydra:v2.0.3
- image of kratos from the latest code from gitlab (cause latest image from dockerhub doesn't have some important fixes)

In short - I can't get "return_to" value because there is no accept oauth2 login request to hydra.

Flow:
1) Request from hydra http://127.0.0.1:4444/oauth2/auth?audience=&client_id=cfe8e0c6-f640-4e3c-9ff1-57c9ec88...... -> GET redirect 300
2) Login challenge: http://localhost:3000/login?login_challenge=c674190877504c69ba2773374a2afe6b - GET 200 OK
3) Init browser flow - http://localhost:4433/self-service/login/browser?refresh=false&login_challenge=c674190877504c69ba2773374a2afe6b - GET 200 OK, returns oauth2_login_challenge/oauth2_login_request (but should it return on this stage?)
![image](https://user-images.githubusercontent.com/97601723/231126991-2f60ed68-babf-4add-8c58-1bc4e6825f02.png)
4) After pressing the login button, we get:
http://localhost:4433/self-service/login?flow=6e6f427f-0297-4236-bc09-92404fc683ad - OPTIONS 204 No content
http://localhost:4433/self-service/login?flow=6e6f427f-0297-4236-bc09-92404fc683ad - POST 200 OK and standard kratos session returned, but hydra should be called  with a PUT request after success login
The reason lies in the logic of kratos - the request sends json, which is why the post login hook returns  at the stage of checking it is json or not.
The official documentation says that it is enough to write "accept text/html" (or any other type, but not json) in headers of updateLoginFlow,  and after testing the result is the same.
In the working versionv(not with our login page), with using kratos-selfservice-ui-node the request content-type  = application/x-www-form-urlencoded. After setting that it does not initiate a PUT request for the hydra, just returns a POST request with a redirect, but has problems with CORS because the updateLoginFlow method cannot parse the data (only json, i suppose) and throws an error "S[] could not find a strategy to login". (don't understand why there is '*' origins, I think it's due to a mistake)
![image](https://user-images.githubusercontent.com/97601723/231127378-817527ef-aeaf-4379-9228-b1ec381c9b58.png)

Trying to comment some logic of checking on json-type in kratos, hydra makes a PUT request, but does not return a response (which is not surprising, but this was only done for checking  that it works at all)
In kratos, as in hydra, all possible origins are registered as allowed.
It seems like flows problem, but i'm not sure.
There is login page code, using ory/client npm "v1.1.24"
Github cant paste it normally, sorry for this, I will add in comments

BTW:
I noticed that the redirection to login flow ( p.4) in the working version of kratos-selfservice-ui-node is carried out BEFORE login, not like ours that inits after. However, I'm not sure if this is the cause of the error.

Kratos API: 
https://www.ory.sh/docs/kratos/reference/api#tag/frontend/operation/updateLoginFlow

If you need to provide  more information, please write about it. Thank you!

**Comment by MiraBlack** [comment]
There is login page logic, change to tsx format.
[LoginPage.txt](https://github.com/ory/kratos/files/11199520/LoginPage.txt)

**Comment by Jorgagu** [comment]
Hi @MiraBlack, there is timaebi who created issue #3225 to say that he can't get the return_to as you say. So I created PR #3251 to fix this problem.
